site/en/blog/shared-element-transitions-for-spas/index.md

```diff
@@ -0,0 +1,196 @@
+  // Transition complete!
+}
+```
+
+To perform transitions, you:
+
+1. Call the `prepare()` function to capture the current visual state.
+1. Update the DOM to represent the new page.
+1. Call `start()` to perform the transition.
+
+`rootTransition` controls the style of the transition.
+
+[Here's a live demo of the styles that are currently available](https://root-transitions-demo.glitch.me/).
+
+#### Limitations of simple transitions
+
+**Animations suddenly stop on the outgoing DOM.** The outgoing DOM is captured as a single frame, meaning that things like gifs and CSS animations will freeze. This is by design, and unlikely to change. This only impacts the outgoing DOM; the new DOM will animate as usual.
+
+**The transition applies to the whole document.** You can't limit the transition to some inner-UI. This feature may be added in future.
+
+**The naming is currently directional, for example, `cover-left`.** This will be changed to match [CSS logical properties](https://web.dev/learn/css/logical-properties/), so `start` and `end` will be used rather than `left` and `right`, and `inline` and `block` will be used to refer to the axis, allowing the API to react to the writing mode of the page.
+
+**Control over the transition is limited.** You can't control the length or other properties of the transition. This will change in future versions of the API, although this will always be limited to things that can be done with a screenshot of state, such as transforms, opacity, and crops.
+
+### Shared element transitions
+
+Shared element transitions build on top of simple transitions by letting you specify elements to transition independently. This is how the header in the following example stays static during these transitions:
+
+{% Video
+  src="video/CZmpGM8Eo1dFe0KNhEO9SGO8Ok23/63bGGDsvP2sE7b009F6w.mp4",
+  loop=true,
+  muted=true,
+  playsinline=true,
+  controls=true
+%}
+
+But they don't need to stay static. They can also move independently of the root transition.
+
+Here's how it's done:
+
+```js
+// When the user clicks on a link/button:
+async function navigateToSettingsPage() {
+  // Capture and visually freeze the current state.
+  await document.documentTransition.prepare({
+    rootTransition: 'cover-up',
+    sharedElements: [element1, element2, element3],
+  });
+  // This is a function within the web app:
+  updateDOMForSettingsPage();
+  // Start the transition.
+  await document.documentTransition.start({
+    sharedElements: [element1, element4, element5],
+  });
+  // Transition complete!
+}
+```
+
+`sharedElements` is the difference here, both in the prepare step and the start step.
+
+In this case, `element1` will animate independently from the rest of the page, moving from its original position & size to its end position & size (or staying still if it doesn't move).
+
+The elements don't need to be the same node. In this case, `element2` will transition to `element4`, because they're each 2nd in the provided array.
```

Comment:
```suggestion
The elements don't need to be the same node. In this case, `element2` will transition to `element4`, because they're each second in the provided array.
```